ServiceDesk/ServiceDeskClient.cs

```diff
@@ -0,0 +1,92 @@
+﻿namespace ServiceDesk
+{
+    using ServiceDesk.Abstractions;
+    using ServiceDesk.Abstractions.DTOs;
+    using System.Collections.Generic;
+    using System.Net.Http;
+    using System.Threading.Tasks;
+    using System.Text.Json;
+    using System.Threading;
+
+    public class ServiceDeskClient : IServiceDeskClient
+    {
+        private const string requestTypesUrl = "https://arcadia-servicedesk-dev31.azurewebsites.net/api/intra/requestTypes";
+
+        private const string currentRequestsUri = "https://arcadia-servicedesk-dev31.azurewebsites.net/api/intra/requests?username=";
+
+        private readonly ServiceDeskRequestPriorityDTO[] priorities = new[] {
+
+                new ServiceDeskRequestPriorityDTO
+                {
+                    Key = 1,
+                    Value="Low"
+                },
+                new ServiceDeskRequestPriorityDTO
+                {
+                    Key = 2,
+                    Value="Default"
+                },
+                new ServiceDeskRequestPriorityDTO
+                {
+                    Key = 3,
+                    Value="High"
+                },
+            };
+
+        public ServiceDeskClient(IHttpClientFactory clientFactory)
+        {
+            this.ClientFactory = clientFactory;
+        }
+
+        private IHttpClientFactory ClientFactory { get; }
+
+        public async Task<IEnumerable<ServiceDeskRequestTypeDTO>> GetRequestTypes(CancellationToken cancellationToken)
+        {
+            var httpRequest = new HttpRequestMessage(HttpMethod.Get, requestTypesUrl);
+
+            httpRequest.Headers.Add("x-api-key", "not-installed");
```

Comment:
> To not do these all every time you want to get or post something to Service Desk I said you to create a service desk client class which will implement all possible interactions with Service Desk API.

And what has changed with your implementation? You anyway copy-pasted same code twice.

Reply:
Added generic method